operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -51,49 +51,38 @@ protected Parameters doInvoke(FHIROperationContext operationContext, Class<? ext
-                resource = resourceHelper.doRead(resourceTypeName, logicalId).getResource();
-                if (resource == null) {
-                    throw buildExceptionWithIssue(resourceTypeName + " with id '" + logicalId + "' was not found", IssueType.NOT_FOUND);
-                }
-            } else if (resourceParameter == null) {
-                throw buildExceptionWithIssue("Input parameter 'resource' is required for the $validate operation", IssueType.INVALID);
-            } else {
-                resource = resourceParameter.getResource();
-            }
+            Parameter modeParameter = getParameter(parameters, "mode");
+            ModeType modeType = getModeType(modeParameter);
+            // Validate the resource parameter
+            Resource resource = validateResource(operationContext, logicalId, resourceHelper, resourceParameter, resourceTypeName, modeType);
             
             List<OperationOutcome.Issue> issues;
             Parameter profileParameter = getParameter(parameters, "profile");
-            Parameter modeParameter = getParameter(parameters, "mode");
             
-            // Only valid "mode" codes are accepted by the $validate operation.
-            validateModeParameter(modeParameter);
+            // Validate mode parameter.
+            validateModeParameter(modeType, resourceHelper, resourceTypeName, operationContext, logicalId);
 
             if (profileParameter != null && profileParameter.getValue() != null) {
                 // If the 'profile' parameter is specified, always validate the resource against that profile.
                 Uri profileUri = profileParameter.getValue().as(Uri.class);
                 String profile = profileUri == null ? null : profileUri.getValue();
                 issues = FHIRValidator.validator().validate(resource, profile);
-            } else if (modeParameter != null && modeParameter.getValue() != null
-                    && ("create".equals(modeParameter.getValue().as(Code.class).getValue())
-                            || "update".equals(modeParameter.getValue().as(Code.class).getValue()))) {
+            } else if (modeType != null
+                    && (modeType == ModeType.CREATE
+                            || modeType == ModeType.UPDATE)) {
                 // If the 'mode' parameter is specified and its value is 'create' or 'update', validate the resource
                 // against the FHIR server config's profile properties and the resource's asserted profiles.
                 issues = resourceHelper.validateResource(resource);
-            } else if (modeParameter != null 
-                    && modeParameter.getValue() != null
-                    && ModeType.DELETE.value().equals(modeParameter.getValue().as(Code.class).getValue())
-                    && FHIROperationContext.Type.INSTANCE.equals(operationContext.getType())) {
+            } else if (modeType != null
```

Comment:
Strictly speaking modeType != null isn't required here because if modeType == DELETE, we know it isn't null.

Reply:
This has been fixed as suggested.